schema/gen/go/genMinima.go

```diff
@@ -3,34 +3,40 @@ package gengo
 import (
 	"fmt"
 	"io"
-)
 
-// EmitMinima emits common code shared by all types --
-// only needs to be output once per module
-func EmitMinima(packageName string, f io.Writer) {
-	// Write header and imports.
-	fmt.Fprintf(f, "package %s\n\n", packageName)
-	f.Write([]byte(`import (
-	ipld "github.com/ipld/go-ipld-prime"
+	wish "github.com/warpfork/go-wish"
 )
-`))
-
-	// Iterator rejection thunks.
-	f.Write([]byte(`
-type mapIteratorReject struct{ err error }
-type listIteratorReject struct{ err error }
-
-func (itr mapIteratorReject) Next() (ipld.Node, ipld.Node, error) { return nil, nil, itr.err }
-func (itr mapIteratorReject) Done() bool                          { return false }
-
-func (itr listIteratorReject) Next() (int, ipld.Node, error) { return -1, nil, itr.err }
-func (itr listIteratorReject) Done() bool                    { return false }
-`))
-
-	// Box type for map keys.
-	// f.Write([]byte(`
-	// type boxedString struct { x string }
-	// `))
-	//
-	// ... nevermind; we already need strings in the prelude.  Use em.
+
+// EmitInternalEnums creates a file with enum types used internal.
+// For example, the state machine values used in map and list builders.
+func EmitInternalEnums(packageName string, w io.Writer) {
+	fmt.Fprint(w, wish.Dedent(`
+		package `+packageName+`
+
+		import (
+			"github.com/ipld/go-ipld-prime/schema"
+		)
+
+		// Code generated go-ipld-prime DO NOT EDIT.
+
+		const (
+			// The 'Maybe' enum does double-duty in this package as a state machine for assembler completion.
```

Comment:
If trying to review and understand how the whole child-completion signalling system works, this couple of comments is kinda key.

TODO(self): this comment block probably shouldn't be emitted into the *output*, though...